### ISSUE TITLE
build: docker build: fatal: Not a git repository

### ISSUE THREAD

**Issue by mjibson** [cockroachdb/cockroach#18952]
On current master, from an Ubuntu machine:

```
$ ./build/builder.sh make clean TYPE=release-linux-gnu
GOPATH set to /go
Running make with -j4
rm -rf /go/native/x86_64-unknown-linux-gnu/jemalloc && git -C /go/src/github.com/cockroachdb/cockroach/c-deps/jemalloc clean -dxf
fatal: Not a git repository: /home/mjibson/src/github.com/cockroachdb/cockroach/.git/modules/c-deps/jemalloc
build/common.mk:447: recipe for target 'clean-c-deps' failed
make: *** [clean-c-deps] Error 128
```

Same error with `make build`.

**Comment by benesch** [comment]
This smells like a git bug, or at least a corrupted Git submodule. We've
had problems with this in the past. You might have some luck with blowing
away the entire Git clone and starting again.

**Comment by mjibson** [comment]
That appeared to fix it.

**Comment by benesch** [comment]
:tada: would you mind posting the version of Git on that box? I want to see if I can find an upstream issue somewhere. It's not yet clear to me what causes the broken submodules.

**Comment by mjibson** [comment]
2.7.4

**Comment by benesch** [comment]
Ah, that explains it. I'll bet it was this bug, which was fixed in 2.8, again: https://github.com/git/git/blob/cf11a67975b057a144618badf16dc4e3d25b9407/Documentation/RelNotes/2.8.3.txt#L33-L36
